individual-files/Cart.java

```diff
@@ -0,0 +1,52 @@
+package sessions;
+
+import javax.servlet.http.*;
+import java.util.Vector;
+import java.util.Enumeration;
+
+public class Cart {
+    Vector v = new Vector();
+    String submit = null;
+    String item = null;
+   
+    private void addItem(String name) {
+        v.addElement(name);
+    }
+
+    private void removeItem(String name) {
+        v.removeElement(name);
+    }
+
+    public void setItem(String name) {
+        item = name;
+    }
+   
+    public void setSubmit(String s) {
+        submit = s;
+    }
+
+    public String[] getItems() {
+        String[] s = new String[ v.size()];
+        v.copyInto(s);
+        return s;
+    }
+   
+    public void processRequest(HttpServletRequest request) {
+        if (submit == null)
+            addItem(item);
+
+        if (submit.equals("add"))
+            addItem(item);
+        else if (submit.equals("remove"))
+            removeItem(item);
+   
```

Comment:
Code style inconsistency. After checking `if (submit == null) addItem(item);` there is no else or else if clause, but another check is done on the submit variable with an else clause